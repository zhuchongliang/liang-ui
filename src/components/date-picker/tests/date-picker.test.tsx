import useDatePicker from "..";
import React, { useState, FC, useEffect } from "react";
import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import "@testing-library/jest-dom";

afterEach(cleanup);

describe("DatePicker", () => {
	const App: FC = function() {
		const [DatePicker, onShow] = useDatePicker();
		const [precision] = useState<"day">("day");
		const [value] = useState<Date>(new Date());
		useEffect(() => {
			onShow();
			// eslint-disable-next-line
		}, []);
		return <DatePicker value={value} precision={precision}/>;
	};
	// test("render DatePicker component", () => {
	// 	const { asFragment } = render(<App />);
	// 	expect(asFragment()).toMatchSnapshot();
	// });
	test("cancel button click", async() => {
		const user = userEvent.setup();
		render(<App />);

		expect(screen.getByText("取消")).toBeInTheDocument();

		await user.click(screen.getByText("取消"));

		expect(screen.getByText("取消")).not.toBeInTheDocument();
	});
});