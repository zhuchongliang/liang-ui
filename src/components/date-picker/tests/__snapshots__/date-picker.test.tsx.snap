// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker render DatePicker component 1`] = `
<DocumentFragment>
  <div
    class="date-picker-component show"
  >
    <div
      class="date-picker-component-mask"
    />
    <div
      class="date-picker-component-body"
    >
      <div
        class="picker-view"
      >
        <div
          class="picker-view-header"
        >
          <a
            class="picker-view-header-button"
            href="#"
          >
            取消
          </a>
          <h3
            class="picker-view-header-title"
          >
            时间选择
          </h3>
          <a
            class="picker-view-header-button"
            href="#"
          >
            确定
          </a>
        </div>
        <div
          class="picker-view-body"
        >
          <div
            class="picker-view-body-column"
          >
            <div
              class="picker-view-body-column-wheel"
              style="transform: none;"
            >
              <div
                class="picker-view-body-column-item"
              >
                2012
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2013
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2014
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2015
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2016
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2017
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2018
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2019
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2020
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2021
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2022
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2023
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2024
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2025
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2026
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2027
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2028
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2029
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2030
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2031
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2032
              </div>
            </div>
          </div>
          <div
            class="picker-view-body-column"
          >
            <div
              class="picker-view-body-column-wheel"
              style="transform: none;"
            >
              <div
                class="picker-view-body-column-item"
              >
                1
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2
              </div>
              <div
                class="picker-view-body-column-item"
              >
                3
              </div>
              <div
                class="picker-view-body-column-item"
              >
                4
              </div>
              <div
                class="picker-view-body-column-item"
              >
                5
              </div>
              <div
                class="picker-view-body-column-item"
              >
                6
              </div>
              <div
                class="picker-view-body-column-item"
              >
                7
              </div>
              <div
                class="picker-view-body-column-item"
              >
                8
              </div>
              <div
                class="picker-view-body-column-item"
              >
                9
              </div>
              <div
                class="picker-view-body-column-item"
              >
                10
              </div>
              <div
                class="picker-view-body-column-item"
              >
                11
              </div>
              <div
                class="picker-view-body-column-item"
              >
                12
              </div>
            </div>
          </div>
          <div
            class="picker-view-body-column"
          >
            <div
              class="picker-view-body-column-wheel"
              style="transform: none;"
            >
              <div
                class="picker-view-body-column-item"
              >
                1
              </div>
              <div
                class="picker-view-body-column-item"
              >
                2
              </div>
              <div
                class="picker-view-body-column-item"
              >
                3
              </div>
              <div
                class="picker-view-body-column-item"
              >
                4
              </div>
              <div
                class="picker-view-body-column-item"
              >
                5
              </div>
              <div
                class="picker-view-body-column-item"
              >
                6
              </div>
              <div
                class="picker-view-body-column-item"
              >
                7
              </div>
              <div
                class="picker-view-body-column-item"
              >
                8
              </div>
              <div
                class="picker-view-body-column-item"
              >
                9
              </div>
              <div
                class="picker-view-body-column-item"
              >
                10
              </div>
              <div
                class="picker-view-body-column-item"
              >
                11
              </div>
              <div
                class="picker-view-body-column-item"
              >
                12
              </div>
              <div
                class="picker-view-body-column-item"
              >
                13
              </div>
              <div
                class="picker-view-body-column-item"
              >
                14
              </div>
              <div
                class="picker-view-body-column-item"
              >
                15
              </div>
              <div
                class="picker-view-body-column-item"
              >
                16
              </div>
              <div
                class="picker-view-body-column-item"
              >
                17
              </div>
              <div
                class="picker-view-body-column-item"
              >
                18
              </div>
              <div
                class="picker-view-body-column-item"
              >
                19
              </div>
              <div
                class="picker-view-body-column-item"
              >
                20
              </div>
              <div
                class="picker-view-body-column-item"
              >
                21
              </div>
              <div
                class="picker-view-body-column-item"
              >
                22
              </div>
              <div
                class="picker-view-body-column-item"
              >
                23
              </div>
              <div
                class="picker-view-body-column-item"
              >
                24
              </div>
              <div
                class="picker-view-body-column-item"
              >
                25
              </div>
              <div
                class="picker-view-body-column-item"
              >
                26
              </div>
              <div
                class="picker-view-body-column-item"
              >
                27
              </div>
              <div
                class="picker-view-body-column-item"
              >
                28
              </div>
              <div
                class="picker-view-body-column-item"
              >
                29
              </div>
              <div
                class="picker-view-body-column-item"
              >
                30
              </div>
              <div
                class="picker-view-body-column-item"
              >
                31
              </div>
            </div>
          </div>
          <div
            class="picker-view-body-mask"
          >
            <div
              class="picker-view-body-mask-top"
            />
            <div
              class="picker-view-body-mask-middle"
            />
            <div
              class="picker-view-body-mask-bottom"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
